Assets/Scripts/StaminaBar.cs

```diff
@@ -31,6 +34,7 @@ public bool move()
         if(stamina > 0)
         {
             stamina--;
+            staminaBarDisplay.ChangeSprite(stamina, maxStamina);

```

Comment:
I don't see what the problem here is, the stamina bar class updates its displayer object. its a question of what is responsible for doing so, either this has a reference to its display, or the display has a reference to this. either way because they are separate entities they need to have coupling.